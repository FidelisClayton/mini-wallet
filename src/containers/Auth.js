import React, { Component } from 'react'
import { css } from 'emotion'

import Login from '../components/Login'
import Register from '../components/Register'

const styles = css({
  height: '100vh',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  justifyContent: 'center',
  backgroundColor: '#8BC34A',
  textAlign: 'center',
  color: '#DCEDC8',
})

export default class Auth extends Component {
  state = {
    register: false,
    login: true
  }

  register = () => {
    this.setState({
      register: true,
      login: false
    })
  }

  login = () => {
    this.setState({
      register: false,
      login: true
    })
  }

  render () {
    return (
      <div className={`auth ${styles}`}>
        <Login
          active={this.state.login}
          onClick={this.login}
        />

        <Register
          active={this.state.register}
          onClick={this.register}
        />
      </div>
    )
  }
}
