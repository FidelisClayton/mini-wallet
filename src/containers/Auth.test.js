import React from 'react'
import { shallow } from 'enzyme'

import { Auth } from './Auth'

describe('Component: <Auth />', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <Auth
        auth={{ error: null }}
        createUser={jest.fn()}
        checkCredentials={jest.fn()}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
