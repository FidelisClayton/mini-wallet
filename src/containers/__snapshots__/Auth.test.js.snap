// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Auth /> should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Auth
    auth={
        Object {
            "error": null,
          }
    }
    checkCredentials={[Function]}
    createUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="auth__wrapper"
>
        <Login
                active={true}
                error={null}
                onClick={[Function]}
                onSubmit={[Function]}
        />
        <Register
                active={false}
                onClick={[Function]}
                onSubmit={[Function]}
        />
</div>,
      "className": "auth css-ke8v07",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Login
            active={true}
            error={null}
            onClick={[Function]}
            onSubmit={[Function]}
/>,
          <Register
            active={false}
            onClick={[Function]}
            onSubmit={[Function]}
/>,
        ],
        "className": "auth__wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "error": null,
            "onClick": [Function],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "onClick": [Function],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="auth__wrapper"
>
          <Login
                    active={true}
                    error={null}
                    onClick={[Function]}
                    onSubmit={[Function]}
          />
          <Register
                    active={false}
                    onClick={[Function]}
                    onSubmit={[Function]}
          />
</div>,
        "className": "auth css-ke8v07",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Login
              active={true}
              error={null}
              onClick={[Function]}
              onSubmit={[Function]}
/>,
            <Register
              active={false}
              onClick={[Function]}
              onSubmit={[Function]}
/>,
          ],
          "className": "auth__wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": true,
              "error": null,
              "onClick": [Function],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "onClick": [Function],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
